caravel/assets/javascripts/SqlLab/reducers.js

```diff
@@ -13,13 +13,39 @@ const defaultQueryEditor = {
 
 export const initialState = {
   alerts: [],
-  queries: [],
+  queries: {},
   queryEditors: [defaultQueryEditor],
   tabHistory: [defaultQueryEditor.id],
   tables: [],
   workspaceQueries: [],
 };
 
+function removeFromDict(state, arrKey, obj, idKey = 'id') {
```

Comment:
"dictionary" is how python calls its hash maps, javascript just calls them "object"